src/bindings/python/src/openvino/runtime/opset2/ops.py

```diff
@@ -129,31 +130,64 @@ def reorg_yolo(input: Node, stride: List[int], name: Optional[str] = None) -> No
 def roi_pooling(
     input: NodeInput,
     coords: NodeInput,
-    output_size: TensorShape,
-    spatial_scale: NumericData,
-    method: str,
+    output_roi: Optional[TensorShape] = None,
+    spatial_scale: Optional[NumericData] = None,
+    method: str = "max",
     name: Optional[str] = None,
+    *,
+    output_size: Optional[TensorShape] = None,
 ) -> Node:
     """Return a node which produces an ROIPooling operation.
 
     :param input:          Input feature map `{N, C, ...}`.
     :param coords:         Coordinates of bounding boxes.
-    :param output_size:    Height/Width of ROI output features (shape).
+    :param output_roi:     Height/Width of ROI output features (shape).
     :param spatial_scale:  Ratio of input feature map over input image size (float).
-    :param method:         Method of pooling - string: "max" or "bilinear".
+    :param method:         Method of pooling - string: "max" or "bilinear". Default: "max"
+    :param output_size:    (DEPRECATED!) Height/Width of ROI output features (shape).
+                           Will override `output_roi` if used and change behavior of the operator.
     :return:               ROIPooling node.
     """
+    # Allow either one of these attributes to be passed.
+    if output_roi is None and output_size is None:
+        raise AttributeError("One of the following arguments must be defined: `output_roi`, `output_size`!")
+    # Force checking of spatial_scale.
+    if spatial_scale is None:
+        raise AttributeError("The following arguments must be defined: `spatial_scale`!")
+
+    def _deprecated_output_size_arg(output_roi: Optional[TensorShape], output_size: Optional[TensorShape]) -> Optional[TensorShape]:
+        if output_size is not None:
+            warnings.warn(
+                "`output_size` is deprecated and will be removed in future. "
```

Comment:
shouldn't the release version be specified where it will be removed?

Reply:
I don't have powers to do so. It comes from operator itself, also without specified version:
https://github.com/openvinotoolkit/openvino/blob/ccfe58cf847a2d306979cf7cb40da5e74ead767a/src/core/include/openvino/op/roi_pooling.hpp#L45-L49
So I guess that @praasz @mitruska and @ilya-lavrenov should decide here. It can be adjusted later with regards to release schedules.